include/tdnf-common-defines.h

```diff
@@ -33,6 +33,12 @@
         }                                                   \
     } while(0)
 
+#define BAIL_ON_TDNF_SYSTEM_ERROR0(dwError)                  \
```

Comment:
I'm not able to understand its usage. We set the errno to dwError before calling this macro. Why can't we use the BAIL_ON_SYSTEM_ERROR instead ?

Reply:
Sorry, I should have explained that one.

The reason is that in lots of places we set dwError to errno, based on the assumption that `errno != 0`. For example:
```
111  	    fp = fopen(pszFile, "w");
112  	    if (!fp)
113  	    {
114  	        dwError = errno;
115  	        BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
116  	    }
```
`BAIL_ON_TDNF_SYSTEM_ERROR` would then `goto error:`, but only if `dwError != 0`. But coverity does not make that assumption, so in the case that `!fp` is `true` and `errno==0` the code would continue without bailing out, with an invalid `fp`. To fix that, I created the `BAIL_ON_TDNF_SYSTEM_ERROR0` macro that would bail in any case, regardless of the value of dwError (that is even if `errno==0`). It makes sense too, because an error has definitely happened although errno is not set. tdnf would show "unknown system error".

This is what coverity showed before the change:
```
111  	    fp = fopen(pszFile, "w");
(6) Event cond_true:	Condition "!fp", taking true branch.
(7) Event var_compare_op:	Comparing "fp" to null implies that "fp" might be null.
Also see events:	[var_deref_model]
112  	    if (!fp)
113  	    {
114  	        dwError = errno;
(8) Event cond_false:	Condition "dwError", taking false branch.
(9) Event if_end:	End of if statement.
115  	        BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
116  	    }
(10) Event var_deref_model:	Passing null pointer "fp" to "fputs", which dereferences it.
Also see events:	[var_compare_op]
117  	    fputs(data, fp);
118  	    fclose(fp);
119  
```

Reply:
You could just add ERROR_BASE to errno in-place and force it to always take the exit path when `errno == 0` without introducing a new macro. This makes it very obvious to the readers on what you are trying to do. Besides, it's not that much of a common use case.

At the very least, this macro has to be refactored to something else (BAIL_UNCOND pops up immediately) as:
- this is identical to the conditional bail at first glance.
- doens't convey tha lack of check.

Reply:
I like `BAIL_ON_TDNF_SYSTEM_ERROR_UNCOND`, much better than `BAIL_ON_TDNF_SYSTEM_ERROR0`.

Reply:
Yes, `BAIL_ON_TDNF_SYSTEM_ERROR_UNCOND` sounds good.

Reply:
Done.